### ISSUE TITLE
Bun.serve expects idleTimeout to be 255 or less

### ISSUE THREAD

**Issue by Gobd** [oven-sh/bun#15589]
### What version of Bun is running?

1.1.37+8ca0eb831

### What platform is your computer?

Darwin 24.1.0 arm64 arm

### What steps can reproduce the bug?

Pass idleTImeout greater than 255

### What is the expected behavior?

No errors

### What do you see instead?

Bun.serve expects idleTimeout to be 255 or less

**Comment by cirospaciari** [comment]
workaround use `idleTimeout: 0` to disable (not recomended) or reset the timer of the request calling server.timeout(req, timeout) (this will reset the timeout each time you call it) we should increase this range in the future.

**Comment by Gobd** [comment]
Like this?

```typescript
Bun.serve({
    port: port,
    fetch: async (req: Request, server: Server): Promise<Response> => {
        server.timeout(req, 310);
        return this._honoApp.fetch(req);
    },
    maxRequestBodySize: 1024 * 1024 * 5,
    idleTimeout: 0,
});
```

**Comment by cirospaciari** [comment]
> Like this?
> 
> Bun.serve({
>     port: port,
>     fetch: async (req: Request, server: Server): Promise<Response> => {
>         server.timeout(req, 310);
>         return this._honoApp.fetch(req);
>     },
>     maxRequestBodySize: 1024 * 1024 * 5,
>     idleTimeout: 0,
> });

Yes but you are still limited to up to 255 in value, but you can call how many times you want and reset it to extend the timeout

**Comment by exKAZUu** [comment]
We are developing an LLM-powered API server. Recent LLMs with "thinking" capabilities have significantly longer response times. For example, Claude 4 with thinking mode can take over 255 seconds to generate its first token. Given this trend toward more computationally intensive models, extending the timeout limit has become increasingly important.

**Comment by niieani** [comment]
Leaving this here in case someone else bumps into the same issue. - 255 being the odd maximum aside, this setting being 10s as a default is such a hidden foot gun in the days of LLM SSE streaming being the norm. I was running a Bun proxy in front of OpenAI and I kept getting connection drops in the middle of a stream, but only after the token count started creeping up. Because GPT would sometimes stall for >10s during inference and no bytes were sent during that time. Hard to reproduce too.
Spent days debugging the entire stack, load balancers, internal gateways, and client app... 🤦‍♂️ Maybe having this field provided as mandatory would be more user friendly?
